databricks/koalas/mlflow.py

```diff
@@ -0,0 +1,158 @@
+
+    @lazy_property
+    def _return_type(self) -> DataType:
+        hint = self._return_type_hint
+        # The logic is simple for now, because it corresponds to the default
+        # case: continuous predictions
+        # TODO: do something smarter, for example when there is a sklearn.Classifier (it should
+        # return an integer or a categorical)
+        # We can do the same for pytorch/tensorflow/keras models by looking at the output types.
+        # However, this is probably better done in mlflow than here.
+        if hint == 'infer' or not hint:
+            hint = np.float64
+        return as_spark_type(hint)
+
+    @lazy_property
+    def _model(self) -> PythonModel:
+        return load_pyfunc(self._path, self._run_id)
+
+    @lazy_property
+    def _model_udf(self):
+        spark = default_session()
+        return spark_udf(spark, self._path, self._run_id, result_type=self._return_type)
+
+    def __str__(self):
+        return "PythonModelWrapper({})".format(str(self._model))
+
+    def __repr__(self):
+        return "PythonModelWrapper({})".format(repr(self._model))
+
+    def predict(self, data):
+        """
+        Returns a prediction on the data.
+        If the data is a pandas Dataframe, the return is a pandas series (TODO: np.array?)
+        If the data is a koalas DataFrame, the return is a Koalas Series.
+        """
+        if isinstance(data, pd.DataFrame):
+            return self._model.predict(data)
+        if isinstance(data, DataFrame):
+            cols = [data._sdf[n] for n in data.columns]
+            return_col = self._model_udf(*cols)
+            return Series(data=return_col, anchor=data, index=data._metadata.index_map)
+
+
+def load_model(path, run_id=None, predict_type='infer') -> PythonModelWrapper:
+    """
+    Loads an MLflow model into an wrapper that can be used both for pandas and Koalas dataframes.
+
+    Parameters
+    ----------
+    path : the path of the model, as logged when calling mlflow.log_model or mlflow.save_model
+    run_id : the id of the run. See MLflow runs documentation for more details.
+    predict_type : a python basic type, a numpy basic type, a Spark type or 'infer'.
+       This is the return type that is expected when calling the predict function of the model.
+       If 'infer' is specified, the wrapper will attempt to determine automatically the return type
+       based on the model type.
+
+    Returns
+    -------
+    PythonModelWrapper
+        A wrapper around MLflow PythonModel objects. This wrapper is expected to adhere to the
+        interface of mlflow.pyfunc.PythonModel.
+
+    Examples
+    --------
+    Here is a full example that creates a model with scikit-learn and saves the model with
+     MLflow. The model is then loaded as a predictor that can be applied on a Koalas
+     dataframe.
+
+    We first initialize our MLflow environment:
+
+    >>> from mlflow.tracking import MlflowClient
+    >>> import mlflow.sklearn
```

Comment:
`import mlflow`?

Reply:
This model is loaded lazily, so it would not work, you need to import it explicitly

Reply:
module

Reply:
I mean, `import mlflow.sklearn` does not actually import the `sklearn` in the context but just `mlflow`, right?
So I think `import mlflow` is enough here.
It's not a big deal anyway, though.

Reply:
ah, I see what you mean. We need it explicitly.